tests/integration/deliver_grant_funding/routes/test_reports.py

```diff
@@ -1067,6 +1068,104 @@ def test_post_add_to_group(self, authenticated_grant_admin_client, factories, db
+        report = factories.collection.create(grant=grant)
+        db_form = factories.form.create(section=report.sections[0])
+
+        form = GroupDisplayOptionsForm(
+            data={
+                "show_questions_on_the_same_page": "True",
+            },
+        )
+        response = authenticated_grant_admin_client.post(
+            url_for(
+                "deliver_grant_funding.add_question_group_display_options",
+                grant_id=grant.id,
+                form_id=db_form.id,
+            ),
+            data=form.data,
+            follow_redirects=False,
+        )
+        assert response.status_code == 400
+
+    @pytest.mark.parametrize(
+        "client_fixture, can_access",
+        (
+            ["authenticated_grant_member_client", False],
+            ["authenticated_grant_admin_client", True],
+        ),
+    )
+    def test_get(self, request, client_fixture, can_access, factories, db_session):
+        client = request.getfixturevalue(client_fixture)
+        report = factories.collection.create(grant=client.grant, name="Test Report")
+        form = factories.form.create(section=report.sections[0], title="Organisation information")
+
+        response = client.get(
+            url_for(
+                "deliver_grant_funding.add_question_group_display_options",
+                grant_id=client.grant.id,
+                form_id=form.id,
+                name="Test group",
+            )
+        )
+
+        if not can_access:
+            assert response.status_code == 403
+        else:
+            assert response.status_code == 200
+            soup = BeautifulSoup(response.data, "html.parser")
+            assert get_h1_text(soup) == "How should the question group be displayed?"
+
+    def test_post(self, authenticated_grant_admin_client, factories, db_session):
+        grant = authenticated_grant_admin_client.grant
+        report = factories.collection.create(grant=grant, name="Test Report")
+        db_form = factories.form.create(section=report.sections[0], title="Organisation information")
+
+        form = GroupDisplayOptionsForm(
+            data={
+                "show_questions_on_the_same_page": "True",
+            },
+        )
+        response = authenticated_grant_admin_client.post(
+            url_for(
+                "deliver_grant_funding.add_question_group_display_options",
+                grant_id=grant.id,
+                form_id=db_form.id,
+                name="Test group",
+            ),
+            data=form.data,
+            follow_redirects=False,
+        )
+        assert response.status_code == 302
+        assert response.location == AnyStringMatching("/grant/[a-z0-9-]{36}/group/[a-z0-9-]{36}")
+
+        response = authenticated_grant_admin_client.get(response.location)
+        assert response.status_code == 200
+        soup = BeautifulSoup(response.data, "html.parser")
+        assert get_h1_text(soup) == "Test group"
```

Comment:
nit: i'm not really a fan of following redirects in tests for a specific endpoint (here the test class is for `AddQuestionGroup`). If we break the redirected-to endpoint then this also fails. Feels like a bit of unnecessary leaking - presumably we get+test that endpoint in its own class, so this doesn't feel super useful.